tests: threads: fix uninitialized scalar variable

naiyuantian opened the pull request:
In the file variable val is not initialized,
causing the variable stack_ptr, pos, points to uninitialized data.
Initialize the variable val according to the code and commits.

Fixes #37916

Signed-off-by: Naiyuan Tian <naiyuan.tian@intel.com>

maksimmasalski commented:
@naiyuantian You need to add into message body, not into title "Fixes #Github_Issue_Number". Use a Github issue number, not a Coverity issue number.

naiyuantian committed:
tests: threads: fix uninitialized scalar variable

In the file variable val is not initialized,
causing the variable stack_ptr, pos, points to uninitialized data.
Initialize the variable val according to the code and commits.

Fixes #37916

Signed-off-by: Naiyuan Tian <naiyuan.tian@intel.com>

maksimmasalski commented:
@naiyuantian I think Coverity wants you to initialize `val` in its declaration like `uint8_t val=0;`

maksimmasalski commented on tests/kernel/threads/thread_stack/src/main.c:
```diff
@@ -161,6 +161,7 @@ void stack_buffer_scenarios(void)
 	 * First test does direct read & write starting at the estimated
 	 * stack pointer up to the highest addresses in the buffer
 	 */
+	val = *stack_start;
```
As I said in the comment, Coverity wants you to initialize `val` in its declaration like `uint8_t val=0;`

maksimmasalski replied:
As I said in the comment, Coverity wants you to initialize val in its declaration like uint8_t val=0;
